src/main/java/com/newestuser/imen/bgdate/BulgarianDate.java

```diff
@@ -0,0 +1,174 @@
+package com.newestuser.imen.bgdate;
+
+import java.time.Year;
+import java.util.Calendar;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BulgarianDate {
+    private int bulgarianDateDay;
+    private String bulgarianDateMonth;
+    private int bulgarianDateYear;
+    private String bulgarianDateTypeOfYear;
+
+    public BulgarianDate(Date gregorianDate){
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(gregorianDate);
+        this.bulgarianDateDay = findBulgarianDay(calendar.get(Calendar.DAY_OF_MONTH), convertMonth(calendar.get(Calendar.MONTH) +1, calendar.get(Calendar.DAY_OF_MONTH)),calendar.get(Calendar.MONTH) +1);
+        this.bulgarianDateMonth = convertMonth(calendar.get(Calendar.MONTH) + 1, calendar.get(Calendar.DAY_OF_MONTH));
+        this.bulgarianDateYear = calendar.get(Calendar.YEAR)+ 4768;
+        this.bulgarianDateTypeOfYear = findYearType(this.bulgarianDateYear);
+//        int dayOfWeek = calendar.get(Calendar.DAY_OF_WEEK);
+//        int monthOfYear = calendar.get(Calendar.MONTH);
+//        int year = calendar.get(Calendar.YEAR);
+    }
+
+
+    public BulgarianDate() {
+
```

Comment:
you can chain the constructors so that the default constructor instantiates a BulgarianDate for the current day

```java
public BulgarianDate() {
   this(new Date());
}
```

Currently some other part of the code can use this constructor to instantiate a date that would have illegal state.